src/DynamicModel.cpp

```diff
@@ -40,6 +40,31 @@ base::Vector6d DynamicModel::calcAcceleration(const base::Vector6d &control_inpu
     return invert_inertia_matrix*acceleration;
 }
 
+base::Vector6d DynamicModel::calcEfforts(const base::Vector6d& acceleration, const base::Vector6d& velocity, const base::Orientation& orientation) const
+{
+    // Check inputs
+    checkAcceleration(acceleration);
+    checkVelocity(velocity);
+
+    // Calculating the efforts given the current state based on all the hydrodynamics effects
+    base::Vector6d efforts = base::Vector6d::Zero();
+
+    efforts = uwv_parameters.inertia_matrix * acceleration + calcGravityBuoyancy(orientation, uwv_parameters);
+    switch(uwv_parameters.model_type)
```

Comment:
I implemented the caclDampingEffect method. It was already declared but apparently never implemented.